pageserver/src/tenant/mgr.rs

```diff
@@ -262,14 +268,31 @@ pub async fn shutdown_all_tenants() {
     for (tenant_id, tenant) in tenants_to_shut_down {
         join_set.spawn(
             async move {
+                let (_guard, shutdown_progress) = completion::channel();
                 let freeze_and_flush = true;
+                if let Err(progress) = tenant.shutdown(shutdown_progress, freeze_and_flush).await {
+                    drop(_guard);
+                    // there is already something else shutting down the tenant (detach, ignore), lets wait for it
+                    let mut remaining = std::pin::pin!(progress.wait());
+                    let remaining = tokio::select! {
+                        biased;
```

Comment:
One-line comment why `biased` is important / useful here?

Reply:
I had hoped it would be equivalent to:

```rust
if futures::future::poll_immediate(&mut remaining).await.is_some() {
   // arm for the remaining
}

tokio::select! {
  // normal construction here
}
```

but it does not actually do that; the sleep is allocated right away before first poll, but it might still be unregistered..  Yes, it is. So it kinda works, if you'd assume that the `remaining` was already done then the sleep is never registered, but it's not what I was hoping for.

Reply:
biased is no longer there, resolving.